src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -3382,13 +3382,14 @@ private function check_for_page_caching() {
 			$headers['Authorization'] = 'Basic ' . base64_encode( wp_unslash( $_SERVER['PHP_AUTH_USER'] ) . ':' . wp_unslash( $_SERVER['PHP_AUTH_PW'] ) );
 		}
 
+		$caching_url                   = apply_filters( 'site_status_persistent_page_cache_url', home_url( '/' ) );
 		$caching_headers               = $this->get_page_cache_headers();
 		$page_caching_response_headers = array();
 		$response_timing               = array();
 
 		for ( $i = 1; $i <= 3; $i++ ) {
 			$start_time    = microtime( true );
-			$http_response = wp_remote_get( home_url( '/' ), compact( 'sslverify', 'headers' ) );
+			$http_response = wp_remote_get( $caching_url, compact( 'sslverify', 'headers' ) );
```

Comment:
The test us design for home page and set message accordingly.

> Page cache is detected by looking for an active page cache plugin as well as making three requests to the homepage and looking for one or more of the following HTTP client caching response headers:

Reply:
Updated the line with the description. If the filter is not specified for the page being checked, "homepage" will be displayed, otherwise the page specified in the filter.